workflowtrigger/main.go

```diff
@@ -14,43 +14,42 @@ import (
 	"greenops.io/workflowtrigger/api/argo"
 	"greenops.io/workflowtrigger/api/commanddelegator"
 	"greenops.io/workflowtrigger/api/reposerver"
+	"greenops.io/workflowtrigger/apikeysmanager"
 	"greenops.io/workflowtrigger/schemavalidation"
 )
 
 func main() {
-	var dbOperator db.DbOperator
-	var kubernetesClient kubernetesclient.KubernetesClient
-	var kafkaClient kafkaclient.KafkaClient
-	var tlsManager tlsmanager.Manager
-	var repoManagerApi reposerver.RepoManagerApi
-	var commandDelegatorApi commanddelegator.CommandDelegatorApi
-	var argoAuthenticatorApi argo.ArgoAuthenticatorApi
-	var schemaValidator schemavalidation.RequestSchemaValidator
-	kubernetesClient = kubernetesclient.New()
-	tlsManager = tlsmanager.New(kubernetesClient)
-	dbOperator = db.New(starter.GetDbClientConfig())
+	kubernetesClient := kubernetesclient.New()
+	tlsManager := tlsmanager.New(kubernetesClient)
+	dbOperator := db.New(starter.GetDbClientConfig())
 	kafkaClient, err := kafkaclient.New(starter.GetKafkaClientConfig(), tlsManager)
 	if err != nil {
 		log.Fatal(err)
 	}
-	repoManagerApi, err = reposerver.New(starter.GetRepoServerClientConfig(), tlsManager)
+	repoManagerApi, err := reposerver.New(starter.GetRepoServerClientConfig(), tlsManager)
 	if err != nil {
 		log.Fatal(err)
 	}
-	commandDelegatorApi, err = commanddelegator.New(starter.GetCommandDelegatorServerClientConfig(), tlsManager)
+	apikeysManager := apikeysmanager.New(kubernetesClient)
+	if err = apikeysManager.GenerateDefaultKeys(); err != nil {
+		log.Fatal(err)
+	}
+	if err = apikeysManager.WatchApiKeys(); err != nil {
+		log.Fatal(err)
+	}
+	commandDelegatorApi, err := commanddelegator.New(starter.GetCommandDelegatorServerClientConfig(), tlsManager, apikeysManager)
 	if err != nil {
 		log.Fatal(err)
 	}
-	argoAuthenticatorApi = argo.New(tlsManager).GetAuthenticatorApi()
-	schemaValidator = schemavalidation.New(argoAuthenticatorApi, repoManagerApi)
+	argoAuthenticatorApi := argo.New(tlsManager, apikeysManager).GetAuthenticatorApi()
```

Comment:
There should only be 1 core argo object which the rest of the APIs are derived from (otherwise the tokens won't be set correctly when making requests). Right now the argo objects are made "fresh".